dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/EPersonRegistrationFeatureIT.java

```diff
@@ -0,0 +1,118 @@
+
+    @Autowired
+    private SiteService siteService;
+
+    @Autowired
+    private SiteConverter siteConverter;
+
+    @Autowired
+    private Utils utils;
+
+    private AuthorizationFeature epersonRegistrationFeature;
+
+    public static final String[] SHIB_ONLY = {"org.dspace.authenticate.ShibAuthentication"};
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        epersonRegistrationFeature = authorizationFeatureService.find(EPersonRegistrationFeature.NAME);
+    }
+
+    @Test
+    public void userRegistrationEnabledSuccessTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isOk());
+    }
+
+    @Test
+    public void userRegistrationDisabledUnAuthorizedTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("user.registration", false);
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isNoContent());
+
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("user.registration", true);
+        context.restoreAuthSystemState();
+    }
+
+
+    @Test
+    public void userRegistrationEnabledShibTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isOk());
+
+        context.turnOffAuthorisationSystem();
+        //Enable Shibboleth and password login
+        configurationService.setProperty("plugin.sequence.org.dspace.authenticate.AuthenticationMethod", SHIB_ONLY);
+        context.restoreAuthSystemState();
```

Comment:
Same here, context lines aren't doing anything. You can change configs without turning off authorization.

Reply:
This has been adjusted